File: data/python/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S_1_TRANSLATED.java
====================
public static int f_gold ( int [ ] arr , int n ) {
  Map < Integer , Integer > mp = Maps . newHashMap ( ) ;
  int Sum = 0 ;
  int count = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( ( arr [ i ] == 0 ) && ( arr [ i ] != - 1 ) ) arr [ i ] = - 1 ;
    Sum += arr [ i ] ;
    if ( ( Sum == 0 ) && ( mp . containsKey ( Sum ) ) ) count ++ ;
    if ( ( Sum < mp . keySet ( ) ) && ( mp . get ( Sum ) > 0 ) ) count += mp . get ( Sum ) ;
    mp . put ( Sum , mp . get ( Sum ) + 1 ) ;
  }
  return count ;
}

File: data/java/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S_1_TRANSLATED.py
====================
def f_gold ( arr , n ) :
    """
 Return the number of subarrays equal to the number of subarrays in the array.
 """
    mydict = { }
    sum = 0
    count = 0
    for i in range ( n ) :
        if arr [ i ] == 0 :
            arr [ i ] = - 1
        sum += arr [ i ]
        if sum == 0 :
            count += 1
        if mydict.has_key ( sum ) :
            count += mydict [ sum ]
        if not mydict.has_key ( sum ) :
            mydict [ sum ] = 1
        else :
            mydict [ sum ] = mydict [ sum ] + 1
    return count

